src/multivariate/multinomial.jl

```diff
@@ -151,7 +151,7 @@ function _logpdf{T<:Real}(d::Multinomial, x::AbstractVector{T})
         @inbounds xi = x[i]
         @inbounds p_i = p[i]
         s -= R(lgamma(R(xi) + 1))
-        @inbounds s += ifelse((xi==0) & (p_i==0), 0.0, xi * log(p_i)) # log(0^0)=0 not NaN
+        s += ifelse((xi== zero(T)) & (p_i == zero(T)), 0.0, xi * log(p_i)) # log(0^0)=0 not NaN
```

Comment:
Oh sorry, I wasn't clear, I meant
```julia
s += ifelse((xi == 0) & (p_i == 0), zero(T), xi * log(p_i))
```
It's fine to use `zero` in the comparison, but adding `zero(T)` rather than 0.0, which is always `Float64`, avoids any unnecessary type promotion when summing.